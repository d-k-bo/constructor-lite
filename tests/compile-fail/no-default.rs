use constructor_lite::ConstructorLite;

struct DoesNotImplementDefault();

#[derive(ConstructorLite)]
struct Foo {
    #[constructor(default)]
    bar: DoesNotImplementDefault,
    //~^ ERROR mismatched types
}

fn main() {}
