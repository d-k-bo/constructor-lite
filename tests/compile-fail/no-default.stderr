error[E0277]: the trait bound `DoesNotImplementDefault: Default` is not satisfied
 --> tests/compile-fail/no-default.rs:5:10
  |
5 | #[derive(ConstructorLite)]
  |          ^^^^^^^^^^^^^^^ the trait `Default` is not implemented for `DoesNotImplementDefault`
  |
  = note: this error originates in the derive macro `ConstructorLite` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider annotating `DoesNotImplementDefault` with `#[derive(Default)]`
  |
3 | #[derive(Default)]
  |
